core: applications backchannel provider (#5449)
goauthentik/authentik: website/docs/core/applications.md

slug: /applications
---

Applications in authentik are the other half of providers. They exist in a 1-to-1 relationship, each application needs a provider and every provider can be used with one application. Starting with authentik 2023.5, applications can use multiple, to augment the functionality of the main provider, see [Backchannel providers](#backchannel-providers).

> Applications are used to configure and separate the authorization / access control and the appearance in the Library page.

Author: Indeed, good point. I don't think it's called Library anywhere in the UI, we should probably change that on all docs

:::

To give users direct links to applications, you can now use an URL like `https://authentik.company/application/launch/<slug>/`. This will redirect the user directly if they're already logged in, and otherwise authenticate the user, and then forward them.

### Backchannel providers

:::info

> Requires authentik 2023.5

Reviewer: ```suggestion
Requires authentik version 2023.5 or later.
```

Author: We should probably go through all version notes and bring them in the same format

:::

> Backchannel providers can augment the functionality of applications with additional protocols. The main provider of an application provides the SSO protocol used for logging into the application. Backchannel providers can be used for protocols such as [SCIM](../providers/scim/index.md) and [LDAP](../providers/ldap/index.md) to provide directory syncing.

Reviewer: ```suggestion
Backchannel providers can augment the functionality of applications by using additional protocols. The main provider of an application provides the SSO protocol that is used for logging into the application. Then, an additional backchannel provider can be used for protocols such as [SCIM](../providers/scim/index.md) and [LDAP](../providers/ldap/index.md) to provide directory syncing.
```

Author: very slight nit, an application can have multiple backend providers (even of the same kind) hence I used the plural

> Access restrictions configured on an application apply to all its backchannel providers.

Reviewer: ```suggestion
Access restrictions that are configured on an application apply to all of its backchannel providers.
```
